crates/solidity/testing/perf/npm/src/main.rs

```diff
@@ -0,0 +1,185 @@
+        for sut in SubjectUT::iter() {
+            let sut_result = self.run(&path, name, file, sut)?;
+            results.extend(sut_result);
+        }
+
+        Ok(results)
+    }
+
+    fn compute_regex(&self) -> Result<regex::Regex> {
+        if let Some(pattern) = &self.pattern {
+            Ok(regex::Regex::new(pattern)?)
+        } else {
+            Ok(regex::Regex::new(".*")?)
+        }
+    }
+
+    fn individual_file_benchmarks(&self, input_path: &Path, files: &Vec<File>) -> Result<Timings> {
+        let mut results = HashMap::<String, f64>::new();
+
+        let pattern_regex = self.compute_regex()?;
+        for file in files {
+            if !pattern_regex.is_match(&file.name) {
+                continue;
+            }
+
+            let result =
+                self.run_benchmarks(input_path, &file.name, &file.hash, Some(&file.file))?;
+            results.extend(result);
+        }
+        Ok(results)
+    }
+
+    fn project_benchmarks(&self, input_path: &Path, projects: &Vec<Project>) -> Result<Timings> {
+        let mut results = HashMap::<String, f64>::new();
+        let pattern_regex = self.compute_regex()?;
+
+        for project in projects {
+            if !pattern_regex.is_match(&project.name) {
+                continue;
+            }
+
+            let result = self.run_benchmarks(input_path, &project.name, &project.hash, None)?;
+            results.extend(result);
+        }
+        Ok(results)
+    }
+
+    fn run(
+        &self,
+        path: &Path,
+        name: &str,
+        file: Option<&str>,
+        sut: SubjectUT,
+    ) -> Result<Timings, anyhow::Error> {
+        let perf_crate = CargoWorkspace::locate_source_crate("solidity_testing_perf")?;
+        let mut command = Command::new("npx")
+            .arg("tsx")
+            .flag("--trace-uncaught")
+            .flag("--expose-gc")
+            .arg(
+                perf_crate
+                    .join("npm/src/benchmarks/main.mts")
+                    .to_str()
+                    .unwrap(),
+            )
+            .property("--dir", path.to_string_lossy())
+            .property("--name", name);
+
+        if let Some(file) = file {
+            command = command.property("--file", file);
+        }
+
+        command = command
+            .property("--runner", sut.as_ref())
+            .property("--cold", self.cold.to_string())
+            .property("--hot", self.hot.to_string());
+
+        let result = command.evaluate()?;
+
+        match serde_json::from_str(&result) {
+            Ok(output) => Ok(output),
+            Err(e) => {
+                eprintln!("Error: Can't parse output as json.\n\tOutput: {result}");
+                Err(e.into())
+            }
+        }
+    }
+}
+
+fn publish(results: impl Iterator<Item = (String, f64)>) -> Result<()> {
+    let results: HashMap<String, _> = results
+        .map(|(k, v)| {
+            (
+                k,
+                json!({
```

Comment:
I decided to write manually this piece instead of resorting to another dummy struct